src/gameboy/include/gameboy/gameboy.h

```diff
@@ -89,11 +90,13 @@ class Gameboy {
 	                                Video *video_ptr);
 
 	/**
-	 * @brief Construct a new Gameboy object
+	 * @brief Construct a new Gameboy object, use this only in case of a
+	 * debugger set true for recording the controls
 	 *
 	 * @param rom_path Path to ROM File
 	 */
-	Gameboy(std::string rom_path);
+	Gameboy(std::string rom_path, bool is_recording = false,
+	        std::string json_file_name = "test.txt");
```

Comment:
I think passing these parameters here is the right idea, but I think this makes instantiating a "GameBoy" a bit complicated, since it now also requires a bunch of debug options.

It's fine that we have these new debug options like isrecording and jsonfilename, but these can perhaps be part of a struct called say `DebugOptions` instead, so that it's also easy to add new options as the Debugger expands in functionality.

I would also suggest renaming "json_file_name" to something that's more descriptive like "recording_output_json_filename".

As a side note, such struct initialization will be a lot neater with C++20. We'll get [Aggregate Initializers](https://en.cppreference.com/w/cpp/language/aggregate_initialization), which will let us write this like

```csharp
// Say the new constructor prototype looks like
Gameboy(std::string rom_path, DebugOptions debug_options = {})

auto gameboy = Gameboy{
    "Tetris.gb",
    {
        .is_recording = "true",
        .recording_output_json_filename = "recording42.json"
    }
};
```
:sob: :sob:

Reply:
That's a pretty neat idea. I think struct should be good enough for now. For now there can be something like `DebugOptions` struct inside the Gameboy class as you mentioned earlier.

Reply:
Debug options doesn't need to be stored as a member if it's only required during construction time. If it is not required while Gameboy is actually running, it just needs to be part of the constructor signature and not a member in the class.

Reply:
Okay cool!